### DISCUSSION TITLE
Connection pool feature in Django 5.1 beta

### DISCUSSION THREAD

**Discussion by medihack** [procrastinate-org/procrastinate#1085] [Q&A]
I just saw from the [release notes of Django 5.1 beta](https://docs.djangoproject.com/en/dev/releases/5.1/#postgresql-connection-pools) that there will be a new connection pool feature for PostgreSQL. Will Procrastinate play nicely with that? Or must a change be made to support this? (It just came to my mind when reading those release notes).

**Comment by ewjoachim** [comment]
The Django connector doesn't create a pool and [doesn't let you use explicitly use one](https://github.com/procrastinate-org/procrastinate/blob/main/procrastinate/contrib/django/django_connector.py#L65-L68) . The worker (which uses listen/notify which is, from the top of my head, the only feature that might be affected by the pool) doesn't use the Django Connector and instead [recreates its own connection pool](https://github.com/procrastinate-org/procrastinate/blob/main/procrastinate/contrib/django/django_connector.py#L161-L163).

It's worth trying, but I believe there's no reason to believe it will break.

I don't know if there's something specific we need to do in order to benefit from this. I believe it will be transparent when we [request a connection](https://github.com/procrastinate-org/procrastinate/blob/main/procrastinate/contrib/django/django_connector.py#L62). What we could do is to try and reuse Django's pool in the worker. This could have an impact if someone runs Django as an ASGI app next to the worker in the same async process.

These are raw thoughts, Anyone with an opinion is likely to be more relevant than me, so please anyone reading this, go ahead and share yours :)
